namer/consul/src/main/scala/io/buoyant/namer/consul/LookupCache.scala

```diff
@@ -72,21 +106,23 @@ private[consul] class LookupCache(
             closableFuture.flatMap(_.close(deadline))
           }
         }
-
-        cachedCounter.incr()
-        new Activity(observation)
+        Activity(observation)
     })
 
-  def apply(
-    dc: String,
-    key: SvcKey,
-    id: Path,
-    residual: Path
-  ): Activity[NameTree[Name]] = {
-    log.debug("consul lookup: %s %s", dc, id.show)
-    lookupCounter.incr()
-    cachedLookup(dc, key, id, residual)
-  }
+  def apply(servicePath: ConsulPath): Option[Activity[NameTree[Name]]] =
+    servicePath.scheme.map {
+      case PathScheme(dc, key, id, residual) =>
+        val cached = Option(lookupStatus.get(servicePath.raw))
+        val address = cached match {
+          case Some(InstrumentedBind(addr, _)) =>
+            Future(addr)
+          case None =>
+            log.debug("consul lookup: %s %s", dc, id.show)
+            lookupCounter.incr()
+            lookupAddress(servicePath.raw, dc, key)
+        }
+        convertToName(address, id, residual)
```

Comment:
I'm not sure about this refactor. With this change, only the Var[Addr] is cached, the Activity[NameTree[Name.Bound]] is recreated on each call.

Reply:
indeed, that's why the function doing that is memoized (i.e. `convertToName`)

To be honset, I'm not entirely sure if memoization is a good fit for an input parameter of type `Future[InstrumentVar[Addr]]` actually.

Would you like it better if we make 2 actual caches, one for the namer call (holding the `Activity[...]`) and another for the admin endpoint calls (holding the `InstrumentedBind`, and thus the original `InstrumentedVar[Addr]`)?

Reply:
Now that I think of it, are you suggest to instead memoize the "whole LookupCache" call instead of simply `convertToName`?

That wouldn't sound so bad in fact